cmd/agent/app/standalone/setup.go

```diff
@@ -0,0 +1,47 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package standalone
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/cmd/agent/common"
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+// SetupCLI sets up the shared utilities for a standalone CLI command:
+// - config, with defaults to avoid conflicting with an agent process running in parallel
+// - logger
+// and returns the log level resolved from cliLogLevel and defaultLogLevel
+func SetupCLI(loggerName config.LoggerName, confFilePath string, cliLogLevel string, defaultLogLevel string) (string, error) {
```

Comment:
yes good point, and agreed we should aim for one common method setting up everything and returning some kind of instance/client.

I don't think we're there yet because there are some differences between the `check` and `jmx` commands on the level of additional initialization they require before calling `setupAutoConfig`, and I haven't investigated enough to know what would be the side effects of refactoring that part. So with the current code, making `SetupCLI` return a client wouldn't really make sense IMHO, but we should move in this direction.

I'll create a follow-up card on this.